mobile/src/components/Shared/AppBar.tsx

```diff
@@ -0,0 +1,52 @@
+// import react native
+import React, { FC, useContext } from "react";
+import { View } from "react-native";
+
+// import react native paper
+import {
+  Appbar,
+  TouchableRipple,
+  Button,
+  Text,
+  Switch,
+  useTheme,
+  Menu,
+} from "react-native-paper";
+
+// import utils
+import { PrefrencesContext } from "../../utils/constants";
+
+// import styles
+import Colors from "../../styles/Colors";
+
+// define Appbar props interface
+interface AppbarProps {
+  title: string;
+  openDrawer: VoidFunction;
+}
+
+// export Appbar component
+const AppBar: FC<AppbarProps> = ({ title, openDrawer }): JSX.Element => {
+  // use theme
+  const theme = useTheme();
+
+  // use theme context
+  const { toggleTheme, isThemeDark } = useContext(PrefrencesContext);
+
+  return (
+    // Appbar header
+    <Appbar.Header theme={{ colors: { primary: theme?.colors.surface } }}>
+      {/* Appbar content */}
+      <Appbar.Action icon="menu" onPress={() => openDrawer()} />
+      <Appbar.Content title={title} />
+      {/* Dark theme switch */}
+      <Text>Dark</Text>
+      <Switch
+        onValueChange={() => toggleTheme()}
+        color={Colors.accent}
+        value={isThemeDark}
```

Comment:
maybe add a bit of margin to the switch

```suggestion
        value={isThemeDark}
        style={{ marginHorizontal: 5 }}
```

it looks better now:

|         |                                                          old                                                          |                                                          new                                                          |
| :-----: | :-------------------------------------------------------------------------------------------------------------------: | :-------------------------------------------------------------------------------------------------------------------: |
|   iOS   |   ![ios_old](https://user-images.githubusercontent.com/20110076/126861480-a9b002d8-d494-4700-9757-83c3b7422321.png)   |   ![ios_new](https://user-images.githubusercontent.com/20110076/126861479-d037e37d-5afa-4447-883f-5d5c2284581b.png)   |
| Android | ![android_old](https://user-images.githubusercontent.com/20110076/126861362-a3992045-f6f3-41f4-a787-b07902916275.png) | ![android_new](https://user-images.githubusercontent.com/20110076/126861361-6ab99823-7c76-4f7a-8bad-bd7de21bca78.png) |

Reply:
Agreed